packages/store/types/tests/selector-options.lint.ts

```diff
@@ -0,0 +1,95 @@
+/* tslint:disable:max-line-length */
+/// <reference types="@types/jasmine" />
+import { Selector, State, SelectorOptions } from '../../src/public_api';
+import { assertType } from './utils/assert-type';
+
+describe('[TEST]: The SelectorOptions decorator', () => {
+  it('should be valid on on State classes', () => {
+    @State<string>({ name: 'my-state' })
+    @SelectorOptions({ injectContainerState: false })
+    class MyState {}
+  });
+
+  it('should be valid on on State class selectors', () => {
+    @State<string>({ name: 'my-state' })
+    class MyState {
+      @Selector([MyState])
+      @SelectorOptions({ injectContainerState: false })
+      static getMyState(state: string) {
+        return state;
+      }
+    }
+  });
+
+  it('should be valid on on query classes', () => {
+    @State<string>({ name: 'my-state' })
+    class MyState {}
+
+    @SelectorOptions({ suppressErrors: false })
+    class MyStateQueries {
+      @Selector([MyState])
+      static getMyState(state: string) {
+        return state;
+      }
+    }
+  });
+
+  it('should be valid on on query class selectors', () => {
+    @State<string>({ name: 'my-state' })
+    class MyState {}
+
+    class MyStateQueries {
+      @Selector([MyState])
+      @SelectorOptions({ suppressErrors: false })
+      static getMyState(state: string) {
+        return state;
+      }
+    }
+  });
+
+  it('should not be a valid parameter decorator', () => {
+    @State<string>({ name: 'my-state' })
+    class MyState {
+      @Selector([MyState])
+      getMyState(
+        @SelectorOptions({ injectContainerState: false }) // $ExpectError
```

Comment:
decorated field?

Reply:
This is just to make it explicit that it cannot be used to decorate a parameter.
There is no typings restriction that we can do for fields or accessors because both of these match the method decorator signature.

Reply:
ok